tests/test_python.py

```diff
@@ -125,25 +142,32 @@ def test_python_webserver_1(container_per_test, hmodule, port, retry):
     t1 = time.time() - t0
 
     # start of the python http server
-    container_per_test.connection.run_expect(
-        [0], f"timeout --preserve-status 120 python3 -m {hmodule} {port} &"
-    )
+    server = container_per_test.connection.run_expect([0], command)
+
+    t1b = time.time() - t0
 
     # port status inspection with timeout
-    for t in range(retry):
-        time.sleep(1)
-        portstatus = container_per_test.connection.socket(
-            f"tcp://0.0.0.0:{port}"
-        ).is_listening
-        if portstatus:
-            break
+    if len(check) > 0:
+        portstatus = str(port) in server.stdout
+        # retry = 0 # skip the loop
+    else:
+        for t in range(retry):
+            time.sleep(1)
+            portstatus = container_per_test.connection.socket(
+                f"tcp://0.0.0.0:{port}"
+            ).is_listening
+            print(t, time.time() - t0)
+            if portstatus:
+                break
+
+    x = container_per_test.connection.run("ps ax | grep http|grep -v grep")
 
     t2 = time.time() - t0
 
     # check inspection success or timeout
     assert (
         portstatus
-    ), f"Timeout expired:Before start {t1}s After {t} checks {t2}s. Expected port not listening"
+    ), f"Timeout expired:Before start {t1}-{t1b}s After {t} checks {t2}s. Expected port not listening"
```

Comment:
What do you mean by "before N seconds After M checks X seconds."? I don't understand the sentence.

Reply:
sentence changed in the new commit